File: campus-order-service/src/main/java/com/campus/order/entity/Order.java
package com.campus.order.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 订单实体 — order_db.`order` 表
 */
@Data
@TableName("`order`")
public class Order {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 幂等键：唯一请求ID，防止重复下单（DB层有 UNIQUE INDEX） */
    private String requestId;

    private Long userId;             // 用户ID
    private String sourceType;       // 菜品来源：CANTEEN/MERCHANT/MIXED
    private Long dishId;             // 菜品ID
    private String dishName;         // 菜品名称（冗余，方便查询）
    private Integer quantity;        // 数量
    private BigDecimal unitPrice;    // 单价
    private BigDecimal totalAmount;  // 总金额
    private String status;           // 订单状态：PENDING_PAYMENT/PAID/PREPARING/COMPLETED/CANCELLED
    private Long paymentId;          // 关联支付单ID
    private String remark;           // 备注

    @TableLogic
    private Integer deleted;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: campus-frontend/src/api/order.ts
import request from './request'

/** 创建订单 */
export const createOrder = (data: {
  userId: number
  sourceType: string   // CANTEEN 或 MERCHANT
  dishId: number
  quantity: number
  requestId: string    // 前端生成UUID，用于幂等性
}) => request.post('/order/create', data)

/** 查询订单详情 */
export const getOrderDetail = (id: number) => request.get(`/order/${id}`)

/** 查询用户订单列表 */
export const getUserOrders = (userId: number) => request.get(`/order/user/${userId}`)
